src/common/upload.ts

```diff
@@ -215,6 +216,40 @@ export abstract class TurboAuthenticatedBaseUploadService
     });
   }
 
+  public async uploadChunked(
+    params: UploadSignedDataItemParams & {
+      chunkSize?: number;
+      batchSize?: number;
+    },
+  ): Promise<TurboUploadDataItemResponse> {
+    const { batchSize, chunkSize, dataItemSizeFactory, events } = params;
+    const totalBytes = dataItemSizeFactory();
+    const uploader = new ChunkedUploader({
+      http: this.httpService,
+      token: this.token,
+      batchSize,
+      chunkSize,
+      logger: this.logger,
+    });
+    if (events !== undefined) {
+      if (events.onUploadProgress) {
```

Comment:
i think you can just pass these event callbacks to the uploader and have it set up the correct listeners

Reply:
handled in chunked.ts now: 246e3ac